fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java

```diff
@@ -471,62 +472,116 @@ public TDescribeTableResult describeTable(TDescribeTableParams params) throws TE
-                    if (TableType.OLAP.equals(table.getType())) {
-                        OlapTable olapTable = (OlapTable) table;
-                        tableKeysType = olapTable.getKeysType().name().substring(0, 3).toUpperCase();
-                    }
-                    for (Column column : table.getBaseSchema()) {
-                        final TColumnDesc desc =
-                                new TColumnDesc(column.getName(), column.getPrimitiveType().toThrift());
-                        final Integer precision = column.getType().getPrecision();
-                        if (precision != null) {
-                            desc.setColumnPrecision(precision);
-                        }
-                        final Integer columnLength = column.getType().getColumnSize();
-                        if (columnLength != null) {
-                            desc.setColumnLength(columnLength);
-                        }
-                        final Integer decimalDigits = column.getType().getDecimalDigits();
-                        if (decimalDigits != null) {
-                            desc.setColumnScale(decimalDigits);
-                        }
-                        if (column.isKey()) {
-                            // COLUMN_KEY (UNI, AGG, DUP, PRI)
-                            desc.setColumnKey(tableKeysType);
-                        } else {
-                            desc.setColumnKey("");
-                        }
-                        final TColumnDef colDef = new TColumnDef(desc);
-                        final String comment = column.getComment();
-                        if (comment != null) {
-                            colDef.setComment(comment);
-                        }
-                        columns.add(colDef);
-                        // if user set limit, then only return limit size result
-                        if (limit > 0 && columns.size() >= limit) {
-                            break;
-                        }
-                    }
-                }
+                setColumnDesc(columns, table, limit, false, params.db, params.getTable_name());
             } finally {
                 db.readUnlock();
             }
         }
         return result;
     }
 
+    // get describeTable without db name and table name parameter, so we need iterate over
+    // dbs and tables, when reach limit, we break;
+    private void describeWithoutDbAndTable(UserIdentity currentUser, List<TColumnDef> columns, long limit) {
+        Catalog catalog = Catalog.getCurrentCatalog();
+        List<String> dbNames = catalog.getDbNames();
+        boolean reachLimit;
+        for (String fullName : dbNames) {
+            if (!Catalog.getCurrentCatalog().getAuth().checkDbPriv(currentUser, fullName, PrivPredicate.SHOW)) {
+                continue;
+            }
+            String dbName = fullName.indexOf(":") >= 0 ? fullName.substring(fullName.indexOf(":") + 1) : fullName;
```

Comment:
Not so robust?

Reply:
Do you mean "String dbName = fullName.indexOf(":") >= 0 ? fullName.substring(fullName.indexOf(":") + 1) : fullName;" ?
ok, we can can process it in be,  then reuse be processing logic,